src/EventListener/RegistrationSuccessListener.php

```diff
@@ -4,31 +4,35 @@
 
 use App\Event\RegistrationSuccessEvent;
 use App\Service\Mailer\Mailer;
-use App\Service\VerifyEmailService;
 use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use SymfonyCasts\Bundle\VerifyEmail\VerifyEmailHelperInterface;
 
 class RegistrationSuccessListener
 {
     private Mailer $mailer;
     private UrlGeneratorInterface $urlGenerator;
-    private VerifyEmailService $verifyEmail;
+    private VerifyEmailHelperInterface $verifyEmailHelper;
 
     public function __construct(
-        VerifyEmailService $verifyEmail,
+        VerifyEmailHelperInterface $verifyEmailHelper,
         Mailer $mailer,
         UrlGeneratorInterface $urlGenerator
     ) {
         $this->mailer = $mailer;
         $this->urlGenerator = $urlGenerator;
-        $this->verifyEmail = $verifyEmail;
+        $this->verifyEmailHelper = $verifyEmailHelper;
     }
 
     public function onRegistrationSuccess(RegistrationSuccessEvent $event)

```

Comment:
не хватает аннотаций и строгой типизации

```suggestion
    /**
     * @param RegistrationSuccessEvent $event
     * @return void
     */
    public function onRegistrationSuccess(RegistrationSuccessEvent $event): void
```